### ISSUE TITLE
Add activity completion rule when a certificate email is sent

### ISSUE THREAD

**Issue by izendegi** [mdjnelson/moodle-mod_customcert#645]
The automatic completion settings for this activity only give the option to be based on the student having viewed the activity, but we've seen that many students don't view the activity because they also received the certificate by email.

Because of that, many students appear to haven't completed the activity in the activity completion reports even if they've already received the certificate.

Having that into account, it would be great if there were another completion requirement option based on having received the certificate.

**Comment by mdjnelson** [comment]
The plugin contains no logic involving the conditions put on an activity. This is a feature available to all activities via the Moodle API. I guess one solution to this would be to trigger a completion event when it gets email, though I am not too sure if that would work. Do you have the resources to confirm this?

**Comment by izendegi** [comment]
Hi, @mdjnelson 

I'm sorry, I think I haven't explained well what I mean :-/

My point wasn't about access restriction conditions but activity completion conditions, specifically those that can be achieved automatically; currently the custom certificate activity gives only one option of this kind of requirements ("View the activity"), which I believe a standard one:
![irudia](https://github.com/user-attachments/assets/5bbd1b5c-7abb-48d2-ad26-1d632c5115af)

Other activities gives more requirement to choose from depending on each activities' logic, some of them also give a "Receive a grade" requirement and [mod_subcourse](https://moodle.org/plugins/mod_subcourse) for example gives also a "Student must complete the referenced course to complete this activity" requirement
![irudia](https://github.com/user-attachments/assets/862f9b56-2a71-41a0-9ef1-88f235a74509)

So, what I want to propose is to add a new one this kind of requirements to the mod_customcert that depends on the student achieving the certificate in addition to the current "View the activity" requirement, because I believe that this requirement is the most accurate of completing the activity.

Is this way, those places which show the completion status of the activities (like the [activity completion report](https://docs.moodle.org/404/en/Activity_completion_report) or the [completion progress block](https://moodle.org/plugins/block_completion_progress)) would show a more accurate status of the completion of this activity.

This change would be also very helpful if you want to add completing the certificate activity to the course completion criteria or as a access restriction for another activity (currently only those students who have accessed the activity are meeting the criteria, not those who received the certificate).

**Comment by mdjnelson** [comment]
Ah, I see.

We have https://github.com/mdjnelson/moodle-mod_customcert/blob/MOODLE_404_STABLE/view.php#L196 which marks the certificate as viewed which ensures the completion status is set. Perhaps this needs to be done when it is emailed as well. If so, this could probably be defined as a bug.

Thoughts?

**Comment by izendegi** [comment]
Yes, I think that marking that requirement as completed also when the certificate is emailed to the student would work.

Using that approach (changing the current requirement to include all the students that have received the certificate), I would suggest to change also the requirement's name to something more accurate ("Receive the certificate" or something similar).

**Comment by mdjnelson** [comment]
I would have to add more logic to the plugin in order to have another field, which means more things I need to maintain. That one is there due to Moodle having it as a default for activities that dont define anything. The sub course plugin defines extra criteria, see https://github.com/catalyst/moodle-mod_subcourse/blob/MOODLE_404_STABLE/mod_form.php#L173. I renamed this issue to reflect what I will do when it is addressed.

**Comment by mdjnelson** [comment]
Implemented in `MOODLE_502_STABLE` and `main`, so it will be available in the next 5.2 release and when I do the 5.3 release. Thanks! Closing.
